src/components/Graph.react.js

```diff
@@ -190,13 +207,13 @@ export default class PlotlyGraph extends Component {
 
     shouldComponentUpdate(nextProps) {
         return (
-            this.props.id !== nextProps.id ||
+            this.state.id !== nextProps.id ||
             JSON.stringify(this.props.style) !== JSON.stringify(nextProps.style)
         );
     }
 
     componentWillReceiveProps(nextProps) {
-        const idChanged = this.props.id !== nextProps.id;
+        const idChanged = this.state.id !== nextProps.id;
```

Comment:
Since we pull the id from the state everywhere now, shouldn't we update the state to the new props.id value?